core/api/service/chain/impl/chain_api_impl.cpp

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/service/chain/impl/chain_api_impl.hpp"
+
+#include <boost/algorithm/hex.hpp>
+#include "common/hexutil.hpp"
+#include "common/visitor.hpp"
+
+namespace kagome::api {
+  using kagome::primitives::BlockHash;
+  namespace {
+    outcome::result<uint32_t> unhexNumber(std::string_view value) {
+      if (value.substr(0, 2) == "0x") {
+        return unhexNumber(value.substr(2));
+      }
+      OUTCOME_TRY(bytes, common::unhex(value));
+      uint32_t result = 0;
+      for (auto b : bytes) {
+        result *= 10;
+        result += b;
+      }
+
+      return result;
+    }
```

Comment:
I suggest moving this to _core/common/hexutil.http_

Reply:
ok